transport/internal/tls/muxlistener/listener.go

```diff
@@ -181,8 +199,7 @@ func (l *listener) mux(ctx context.Context, conn net.Conn) (net.Conn, error) {
 	c := newConnectionSniffer(conn)
 	isTLS, err := matchTLSConnection(c)
 	if err != nil {
-		l.logger.Error("TLS connection matcher failed", zap.Error(err))
-		return nil, err
+		return nil, fmt.Errorf("TLS connection matcher failed: %w", err)
```

Comment:
Since you are already here, could you make the error message in lowercase?
```suggestion
		return nil, fmt.Errorf("tls connection matcher failed: %w", err)
```

Same for the handshake error below